src/CoreWCF.Primitives/src/CoreWCF/Channels/Message.cs

```diff
@@ -631,7 +631,7 @@ protected virtual string OnGetBodyAttribute(string localName, string ns)
             return null;
         }
 
-        internal void ReadFromBodyContentsToEnd(XmlDictionaryReader reader)
+        public void ReadFromBodyContentsToEnd(XmlDictionaryReader reader)
         {
```

Comment:
Keep internal and copy the static helper method immediately below into ReceivedMessage in WebHttp. Then at the call site to this method, replace with with a call to the static method passing Version.Envelope as Version is a public property on Message

Reply:
This has been addressed.